src/DataScienceTraits.jl

```diff
@@ -92,6 +115,7 @@ scitype(::Type{<:Integer}) = Categorical
 scitype(::Type{<:AbstractChar}) = Categorical
 scitype(::Type{<:AbstractString}) = Categorical
 scitype(::Type{Union{T,Missing}}) where {T} = scitype(T)
+scitype(::Type{<:TimeType}) = Temporal
```

Comment:
Move to before `Union{T,Missing}`:
```suggestion
scitype(::Type{<:TimeType}) = Temporal
scitype(::Type{Union{T,Missing}}) where {T} = scitype(T)
```